src/decode-arp.c

```diff
@@ -0,0 +1,59 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ *
+ * \author Giuseppe Longo <giuseppe@glongo.it>
+ *
+ */
+
+#include "suricata-common.h"
+#include "decode.h"
+#include "decode-arp.h"
+#include "decode-events.h"
+
+#define ETHERNET_HW_TYPE 0x01
+
+int DecodeARP(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p, const uint8_t *pkt, uint32_t len)
+{
+    StatsIncr(tv, dtv->counter_arp);
+
+    if (unlikely(len < ARP_HEADER_LEN)) {
```

Comment:
I think this will still potentially flag valid but unsupported arp packets as invalid, if smaller arp packets can exist. I think the check here should be to check for a minimal size required to reach the hw_type/proto_type fields first, then check those, then if they are eth/ip we check the full 28 byte size and reject as invalid if data doesn't have that size.

Reply:
Additionally, what happens if size > 28? Should this give an event?

Another point: the header uses a fixed size for mac and ip, which seems fine, but it does not check if those match the hw_size and proto_size. If there is a mismatch there, it's probably also worth an event.

Reply:
> Additionally, what happens if size > 28? Should this give an event?
> 

At this point, after verifying the minimum length for an ARP packet, could we check that the full length is exactly 28 bytes and, if necessary, issue an event like `unsupported length` (or `packet`) for other packets?

> Another point: the header uses a fixed size for mac and ip, which seems fine, but it does not check if those match the hw_size and proto_size. If there is a mismatch there, it's probably also worth an event.

Agree.